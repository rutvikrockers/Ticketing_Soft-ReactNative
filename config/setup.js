// create our store for testing redux components
import configureMockStore from 'redux-mock-store';
const thunk = require('redux-thunk');

// Setup adapter to work with enzyme 3.2.0
const Enzyme = require('enzyme');
const Adapter = require('enzyme-adapter-react-16');

export const mockStore = configureMockStore([thunk.default]);

Enzyme.configure({ adapter: new Adapter() });