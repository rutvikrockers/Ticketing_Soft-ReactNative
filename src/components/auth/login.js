import React, { Component } from 'react';
import {
  StyleSheet,
  Text,
  TextInput,
  TouchableOpacity,AsyncStorage,
  View
} from 'react-native';
import {Redirect, Link} from 'react-router-native';
import {authenticate,getGuestToken} from '../../actions';
import {connect} from 'react-redux';
import Button from '../shared/button';
import styles from '../../styles/main.js';

class Login extends Component {
  constructor(props) {
    super(props);
    this.state = {email: '', password: ''}
  }
  
   login() {
    this.props.dispatch(authenticate(this.state.email, this.state.password)); 
  }
  render() {
   
    if(this.props.user.loggedIn){
      return ( <Redirect to="/home" /> )
    }
  
    return (
      <View style={styles.container}>
        <View style={styles.inputWrapper}>
          <TextInput
            style={styles.input}
            placeholder="Your Email"
            onChangeText={(email) => this.setState({email})}
            value={this.state.email}
            underlineColorAndroid='rgba(0,0,0,0)'
            autoCapitalize='none'
          />
        </View>
        <View style={styles.inputWrapper}>
          <TextInput
            style={styles.input}
            placeholder="Password"
            autoCapitalize='none'
            onChangeText={(password) => this.setState({password})}
            value={this.state.password}
            underlineColorAndroid='rgba(0,0,0,0)'
            secureTextEntry={true}/>
        </View>
        <View style={{marginTop: 20}} />
        <Button
          style={styles.btnContainer}
          onPress={() => {this.login()}}
          value="LOGIN"
        />
       
      </View>
    );
  }
}

const mapStateToProps = (state) => {
  return {
    error: state.error,
    user: state.user,
  
  }
}

export default connect(mapStateToProps)(Login);
