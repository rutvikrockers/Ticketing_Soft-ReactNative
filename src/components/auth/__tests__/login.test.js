import React from 'react';
import { shallow } from 'enzyme';
import { Provider } from 'react-redux';
import { mockStore } from '../../../../config/setup';
import toJson from 'enzyme-to-json';
import { default as Login } from '../login';

const render = (props) => shallow(
    <Provider store={mockStore({})}>
        <Login {...props} />
    </Provider>
);

describe('Login', () => {
    let component;
    beforeEach(() => {
       component = render({});
    });

    it('matches its snapshot', () => {
       expect(toJson(component, { mode: 'deep' })).toMatchSnapshot();
    });

    it('renders', () => {
       expect(component.find(Login).length).toBe(1);
    });
});