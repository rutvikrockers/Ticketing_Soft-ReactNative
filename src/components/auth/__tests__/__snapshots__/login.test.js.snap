// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login matches its snapshot 1`] = `<Connect(Login) />`;
